tests/cnf/core/network/sriov/tests/rdmametrics.go

```diff
@@ -0,0 +1,368 @@
+					[]string{"bash", "-c", fmt.Sprintf("ls /sys/bus/pci/devices/%s/infiniband/%s/ports/1/hw_counters",
+						pciAddress, rdmaDevice)})
+				Expect(err).To(HaveOccurred(), "Expected command to fail as the path is not present on Pod")
+				Expect(podOutput.String()).To(ContainSubstring("No such file or directory"),
+					fmt.Sprintf("Expected error 'No such file or directory' in the output %s", podOutput.String()))
+
+				By("Verify Rdma Metrics should be present on Host")
+				nodeOutput, err := cluster.ExecCmdWithStdout(APIClient,
+					fmt.Sprintf("ls /sys/bus/pci/devices/%s/infiniband/%s/ports/1/hw_counters", pciAddress, rdmaDevice),
+					metav1.ListOptions{LabelSelector: fmt.Sprintf("kubernetes.io/hostname=%s", workerNodeList[0].Object.Name)})
+				Expect(err).ToNot(HaveOccurred(),
+					fmt.Sprintf("Failed to run command %s on node %s",
+						fmt.Sprintf("ls /sys/bus/pci/devices/%s/infiniband/%s/ports/1/hw_counters",
+							pciAddress, rdmaDevice), testPod.Object.Spec.NodeName))
+				Expect(nodeOutput[fmt.Sprintf(testPod.Object.Spec.NodeName)]).To(ContainSubstring("out_of_buffer"),
+					fmt.Sprintf("Failed to find the counters in the output %s", nodeOutput[testPod.Object.Spec.NodeName]))
+			})
+
+			AfterAll(func() {
+				By("Removing SR-IOV configuration")
+				err := sriovenv.RemoveSriovConfigurationAndWaitForSriovAndMCPStable()
+				Expect(err).ToNot(HaveOccurred(), "Failed to remove SR-IOV configration")
+
+				By("Cleaning test namespace")
+				err = namespace.NewBuilder(APIClient, tsparams.TestNamespaceName).CleanObjects(
+					netparam.DefaultTimeout, pod.GetGVR())
+				Expect(err).ToNot(HaveOccurred(), "Failed to clean test namespace")
+			})
+		})
+	})
+
+func enableRdmaMode() *mco.MCBuilder {
+	RdmaMachineConfigContents := `
+		[Unit]
+		Description=RDMA exclusive mode
+		Before=kubelet.service crio.service node-valid-hostname.service
+		
+		[Service]
+		# Need oneshot to delay kubelet
+		Type=oneshot
+		ExecStart=/usr/bin/bash -c "rdma system set netns exclusive"
+		StandardOutput=journal+console
+		StandardError=journal+console
+		
+		[Install]
+		WantedBy=network-online.target`
+	truePointer := true
+	ignitionConfig := ignition.Config{
+		Ignition: ignition.Ignition{
+			Version: "3.1.0",
+		},
+		Systemd: ignition.Systemd{
+			Units: []ignition.Unit{
+				{
+					Enabled:  &truePointer,
+					Name:     "rdma.service",
+					Contents: &RdmaMachineConfigContents,
+				},
+			},
+		},
+	}
+
+	finalIgnitionConfig, err := json.Marshal(ignitionConfig)
+	Expect(err).ToNot(HaveOccurred(), "Failed to serialize ignition config")
+
+	createdMC, err := mco.NewMCBuilder(APIClient, "02-rdma-netns-exclusive-mode").
+		WithLabel("machineconfiguration.openshift.io/role", "workercnf").
+		WithRawConfig(finalIgnitionConfig).
+		Create()
+	Expect(err).ToNot(HaveOccurred(), "Failed to create RDMA machine config")
+
+	return createdMC
+}
+
+func getInterfacePci(tPod *pod.Builder, podInterface string) (string, string, error) {
```

Comment:
can we reuse getPCIAddressListFromSrIovNetworkName for rootless.go?

Reply:
We need **_rdma-device_** along with **_pci-address_**, but rootless function gives only pci addresses.